src/ray/gcs/gcs_server/gcs_placement_group_scheduler.cc

```diff
@@ -237,14 +237,18 @@ void GcsPlacementGroupScheduler::CancelResourceReserve(
         RAY_LOG(DEBUG) << "Finished cancelling the resource reserved for bundle: "
                        << bundle_spec->DebugString() << " at node " << node_id;
         std::vector<std::string> resource_names;
-        rpc::NodeResourceChange node_resource_change;
         auto &resources = bundle_spec->GetFormattedResources();
         for (const auto &iter : resources) {
           resource_names.push_back(iter.first);
-          node_resource_change.add_deleted_resources(iter.first);
         }
-        node_resource_change.set_node_id(node_id.Binary());
-        ray_syncer_.Update(std::move(node_resource_change));
+        if (ray_syncer_ != nullptr) {
+          rpc::NodeResourceChange node_resource_change;
+          for (const auto &iter : resources) {
+            node_resource_change.add_deleted_resources(iter.first);
+          }
+          node_resource_change.set_node_id(node_id.Binary());
+          ray_syncer_->Update(std::move(node_resource_change));
```

Comment:
Why does it use syncer directly instead of resource manager btw?

Reply:
@rkooo567 here it's the old implementation of ray syncer where pg is sent from GCS.
The integration one will receive the message from syncer.

Which means, if we migrated to the new one, these code will be deleted.

Here:
```
        if (ray_syncer_ != nullptr) {
```